dapp/src/components/dashboard/containers/MainDashboard/index.jsx

```diff
@@ -35,29 +36,44 @@ class DashboardLayout extends Component {
     open: false
   }
 
+  // componentDidMount () {
+  //   const { token } = this.props
+  //   if (!token) {
+  //     const { fetchCommunity, fetchTokenProgress, fetchEntities, communityAddress } = this.props
+  //     fetchCommunity(communityAddress)
+  //     fetchTokenProgress(communityAddress)
+  //     fetchEntities(communityAddress)
+  //   }
+  // }
+
   componentDidMount () {
-    const { token } = this.props
-    if (!token) {
-      const { fetchCommunity, fetchTokenProgress, fetchEntities, communityAddress } = this.props
-      fetchCommunity(communityAddress)
-      fetchTokenProgress(communityAddress)
-      fetchEntities(communityAddress)
+    const { community, communityAddress } = this.props
+    if (community) {
+      this.props.fetchToken(community.homeTokenAddress)
```

Comment:
We have watcher that fetch the token after "FETCH_COMMUNITY_DATA_SUCCESS" called "watchCommunityFetch", you can use it